libs/langchain/langchain/chains/pebblo_retrieval/base.py

```diff
@@ -211,3 +284,171 @@ async def _aget_docs(
+        try:
+            pebblo_resp = requests.post(
+                app_discover_url, headers=headers, json=qa_payload.dict(), timeout=20
+                )
+            print("prompt-payload", qa_payload)
+            logger.debug(
+                "send_prompt[local]: request url %s, body %s len %s\
+                    response status %s body %s",
+                pebblo_resp.request.url,
+                str(pebblo_resp.request.body),
+                str(len(pebblo_resp.request.body if pebblo_resp.request.body else [])),
+                str(pebblo_resp.status_code),
+                pebblo_resp.json(),
+            )
+            if pebblo_resp.status_code in [HTTPStatus.OK, HTTPStatus.BAD_GATEWAY]:
+                PebbloRetrievalQA.set_prompt_sent()
+            else:
+                logger.warning(
+                    f"Received unexpected HTTP response code: {pebblo_resp.status_code}"
+                )
+        except requests.exceptions.RequestException:
+            logger.warning("Unable to reach pebblo server.")
+        except Exception as e:
+            logger.warning("An Exception caught in _send_discover: local %s", e)
+
+        if self.api_key:
+            try:
+                headers.update({"x-api-key": self.api_key})
+                pebblo_cloud_url = f"{PEBBLO_CLOUD_URL}{PROMPT_URL}"
+                pebblo_cloud_response = requests.post(
+                    pebblo_cloud_url, headers=headers, json=qa_payload.dict(), timeout=20
+                )
+
+                logger.debug(
+                    "send_prompt[cloud]: request url %s, body %s len %s\
+                        response status %s body %s",
+                    pebblo_cloud_response.request.url,
+                    str(pebblo_cloud_response.request.body),
+                    str(
+                        len(
+                            pebblo_cloud_response.request.body
+                            if pebblo_cloud_response.request.body
+                            else []
+                        )
+                    ),
+                    str(pebblo_cloud_response.status_code),
+                    pebblo_cloud_response.json(),
+                )
+            except requests.exceptions.RequestException:
+                logger.warning("Unable to reach Pebblo cloud server.")
+            except Exception as e:
+                logger.warning("An Exception caught in _send_prompt: cloud %s", e)
+
+    @classmethod
+    def get_chain_details(cls, llm, **kwargs):
+        chain = [
+            {
+                "name": cls.__name__,
+                "model": {
+                    "name": llm.model_name,
+                    "vendor": llm.__class__.__name__
+                    },
+                "vector_dbs": [
+                    {
+                        "name": kwargs['retriever'].vectorstore.__class__.__name__,
+                        "embedding_model": kwargs['retriever'].vectorstore._embeddings.model
```

Comment:
Attribute `_embeddings` is not available on all the vectorstores, it's failing for Pinecone